device/samples/NetStandardMultiplexSample/Program.cs

```diff
@@ -0,0 +1,138 @@
+        //"hostname.azure-devices.net"
+        private const string IoTHubHostName = "<iothubhostname>.azure-devices.net";
+
+        //Number of devices for multiplexing
+        private static int MUX_DEVICES = 4;
+        private static DeviceClient[] deviceClients = new DeviceClient[MUX_DEVICES];
+
+        //Array of device Ids (should have the same entries as defined on MUX_DEVICES):
+        //{"device1", "device2", "device3", "device4"}
+        private static String[] deviceIds = new String[] { "<device1>", "<device2>", "<device3>", "<device4>" };
+
+        //Array of device PrimaryKeys (following the same order as above):
+        //{"PrimaryKey device1", "PrimaryKey device2", "PrimaryKey device3", "PrimaryKey device4"}
+        private static String[] devicePrimaryKeys = new String[] { "<PrimaryKey device1>",
+                                                                    "<PrimaryKey device2>",
+                                                                    "<PrimaryKey device3>",
+                                                                    "<PrimaryKey device4>"    };
+
+        private static int MESSAGE_COUNT = 5;
+        private const int TEMPERATURE_THRESHOLD = 30;
+
+        private static float temperature;
+        private static float humidity;
+        private static Random rnd = new Random();
+
+        static void Main(string[] args)
+        {
+
+            try
+            {
+                MultiplexConnection().Wait();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error in sample: {0}", ex.Message);
+            }
+
+            Console.WriteLine("Exited!\n");
+        }
+
+        static async Task MultiplexConnection()
+        {
+            for (int i = 0; i < 4; i++)
+            {
+                var auth = new DeviceAuthenticationWithRegistrySymmetricKey(deviceIds[i], devicePrimaryKeys[i]);
+                deviceClients[i] = DeviceClient.Create(
+                    IoTHubHostName,
+                    auth,
+                    new ITransportSettings[]
+                    {
+                        //Must specify Amqp_WebSocket_Only or Amqp_Tcp_Only
+                        new AmqpTransportSettings(TransportType.Amqp_WebSocket_Only)
+                        {
+                            //By default, connection pooling is turned off. It must be turned on explicitly using the below settings.
+                            AmqpConnectionPoolSettings = new AmqpConnectionPoolSettings()
+                            {
+                                Pooling = true,
+                                //If you would like to have only one TCP Connection set MaxPoolSize=1
+                                MaxPoolSize = 2
+                            }
+                        }
+                    });
+                await deviceClients[i].OpenAsync();
+            }
+
+            SendEvent().Wait();
+            ReceiveCommands().Wait();
```

Comment:
Why is Send and Receive factored in other methods while open is not.

I would rather have something on the lines of 
```C#

await OpenAllAsync();
await SendAllAsync();
await ReceiveAllAsync();
```

if this is the semantic you are looking for (i.e. all devices must complete the previous operations before they start the next).

I would not recommend doing this in production though: each operation should execute asynchronously. The only ordering required is that a device Opens, Sends then Receives, regardless of other devices do at that time. (Ties in with the OOP comment I had above.)